src/emx/index.ts

```diff
@@ -0,0 +1,212 @@
+    entity: TABLE_MONTHLY_REPEATED,
+    idAttribute: 'AUTO'
+  },
+  {
+    name: 'child_id',
+    dataType: MolgenisDataType.STRING,
+    entity: TABLE_MONTHLY_REPEATED
+  },
+  {
+    name: 'age_months',
+    dataType: MolgenisDataType.INTEGER,
+    entity: TABLE_MONTHLY_REPEATED,
+    rangeMin: 0,
+    rangeMax: 215
+  },
+  {
+    name: 'id',
+    dataType: MolgenisDataType.STRING,
+    entity: TABLE_YEARLY_REPEATED,
+    idAttribute: 'AUTO'
+  },
+  {
+    name: 'child_id',
+    dataType: MolgenisDataType.STRING,
+    entity: TABLE_YEARLY_REPEATED
+  },
+  {
+    name: 'age_years',
+    dataType: MolgenisDataType.INTEGER,
+    entity: TABLE_YEARLY_REPEATED,
+    rangeMin: 0,
+    rangeMax: 17
+  }
+]
+
+const getAttributes = (variableGroups: Variable[][]): Attribute[] => {
+  const variableAttributes: Attribute[] = variableGroups.reduce(
+    (soFar: Attribute[], variables: Variable[]) => [
+      ...soFar,
+      getAttribute(variables[0], getTablename(variables.length))
+    ],
+    []
+  )
+
+  const categoricals: Variable[] = variableGroups
+    .map(it => it[0])
+    .filter(it => it.datatype.id === CatalogueDatatype.CATEGORICAL)
+  const optionTableAttributes = categoricals
+    .map(it => it.variable)
+    .map(attributeNameOfVariable)
+    .map(key => `${key}_options`)
+    .flatMap(getOptionTableAttributes)
+
+  return [...initialAttributes, ...variableAttributes, ...optionTableAttributes]
+}
+
+const getAttribute = (variable: Variable, entity: string): Attribute => {
+  const name = attributeNameOfVariable(variable.variable)
+  switch (variable.datatype.id) {
+    case CatalogueDatatype.CATEGORICAL:
+      return {
+        name,
+        dataType: MolgenisDataType.CATEGORICAL,
+        description: variable.label,
+        entity,
+        refEntity: `${name}_options`
+      }
+    case CatalogueDatatype.INTEGER:
+      return {
+        name,
+        dataType: MolgenisDataType.INTEGER,
+        description: variable.label,
+        entity
+      }
+    case CatalogueDatatype.BINARY:
+      return {
+        name,
+        dataType: MolgenisDataType.BOOLEAN,
+        description: variable.label,
+        entity
+      }
+    case CatalogueDatatype.CONTINUOUS:
+      return {
+        name,
+        dataType: MolgenisDataType.DECIMAL,
+        description: variable.label,
+        entity
+      }
+    default:
+      throw new Error(`Unknown data type for ${name}: ${variable.datatype.id}`)
+  }
+}
+
+const getOptionTableAttributes = (entity: string): Attribute[] => [
+  {
+    name: 'id',
+    dataType: MolgenisDataType.INTEGER,
+    entity,
+    idAttribute: 'TRUE'
+  },
+  {
+    name: 'label',
+    dataType: MolgenisDataType.STRING,
+    entity,
+    labelAttribute: 'TRUE'
+  }
+]
+
+export const getTablename = (repeatCount: number) => {
+  let tableName = TABLE_NON_REPEATED
+  if (repeatCount > 100) {
+    tableName = TABLE_MONTHLY_REPEATED
+  } else if (repeatCount > 20) {
+    tableName = TABLE_TRIMESTER_REPEATED
```

Comment:
Ah now I see this should be weekly? Leave it out first. I need to check how many variables weekly could contain. Maybe reintroduce the trimester vars. That is === 3